### PR TITLE
Small improvements to make installation easier

### PR DESCRIPTION
**Pull request by jannisteunissen**
1. Makefile: check whether pkg-config failed
2. README.md: add instructions for MATRIX_NO_E2E, update required libs, how to install as non-root

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -4,9 +4,21 @@ CC=gcc
 LIBS=purple json-glib-1.0 glib-2.0 sqlite3
 
 PKG_CONFIG=pkg-config
-CFLAGS+=$(shell $(PKG_CONFIG) --cflags $(LIBS))
+
+PKG_CFLAGS=$(shell $(PKG_CONFIG) --cflags $(LIBS) || echo "FAILED")
+ifeq ($(PKG_CFLAGS),FAILED)
+$(error "$(PKG_CONFIG) failed")
+else
```
Comment (richvdh) [inline Makefile]: might as well just put `endif` here?

File: Makefile [hunk 2]
```diff
@@ -4,9 +4,21 @@ CC=gcc
 LIBS=purple json-glib-1.0 glib-2.0 sqlite3
 
 PKG_CONFIG=pkg-config
-CFLAGS+=$(shell $(PKG_CONFIG) --cflags $(LIBS))
+
+PKG_CFLAGS=$(shell $(PKG_CONFIG) --cflags $(LIBS) || echo "FAILED")
+ifeq ($(PKG_CFLAGS),FAILED)
+$(error "$(PKG_CONFIG) failed")
```
Comment (richvdh) [inline Makefile]:
```suggestion
    $(error "$(PKG_CONFIG) failed")
```

File: Makefile [hunk 3]
```diff
@@ -4,9 +4,21 @@ CC=gcc
 LIBS=purple json-glib-1.0 glib-2.0 sqlite3
 
 PKG_CONFIG=pkg-config
-CFLAGS+=$(shell $(PKG_CONFIG) --cflags $(LIBS))
+
+PKG_CFLAGS=$(shell $(PKG_CONFIG) --cflags $(LIBS) || echo "FAILED")
```
[the remark this answers is not in the document]
Comment (jannisteunissen) [reply Makefile]: Indeed!

File: Makefile (hunk 1 above)
Comment (jannisteunissen) [reply Makefile]: Yes

File: Makefile (hunk 2 above)
Comment (jannisteunissen) [reply Makefile]: Indents were added

**Comment by jannisteunissen** [comment]
Thanks for the suggestions, I'm not that familiar with Github's pull request interface, but I guess they are now corrected.

**Pull request merged**